src/mokernels/moinput.jl

```diff
@@ -51,7 +75,20 @@ function Base.getindex(inp::MOInput, ind::Integer)
     end
 end
 
-Base.iterate(inp::MOInput) = (inp[1], 1)
-function Base.iterate(inp::MOInput, state)
+function Base.getindex(inp::IsotopicByFeatures, ind::Integer)
+    if ind > 0
+        out_dim = (ind - 1) % inp.out_dim + 1
+        ind = (ind - 1) ÷ inp.out_dim + 1
+        return (inp.x[ind], out_dim::Int)
+    else
+        throw(BoundsError(string("Trying to access at ", ind)))
+    end
```

Comment:
As per https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/pull/310/files#r658653385, I suggest
```suggestion
    @boundscheck checkbounds(inp, ind)
    feature_index, output_index = fldmod1(ind, inp.out_dim)
    feature = @inbounds inp.x[feature_index]
    return feature, output_index
```

The other `getindex` implementation should be updated in a similar way.

Reply:
Thank you for this! It has been implemented in both dispatches :)